pagefaultgames/pokerogue
# Fix - Update UiInputs Class to Use buttonAb and buttonCycleOption Methods like before

## What are the changes?

The changes involve updating the `UiInputs` class to use the `buttonAb` and `buttonCycleOption` methods for specific button actions instead of the previous direct method calls. Like before the small refactor to remove error code from VSCode

## Why am I doing these changes?



These changes are intended to ensure consistency in how button actions are handled.

## What did change?


The following changes were made:
- Replaced `this.buttonDirection` calls with `this.buttonAb` for `Button.ACTION` and `Button.CANCEL`.
- Replaced individual `buttonDirection` calls for various `Button.CYCLE_*` actions with `this.buttonCycleOption`.

Before:
```typescript
[Button.ACTION]: () => this.buttonDirection(Button.ACTION),
[Button.CANCEL]: () => this.buttonDirection(Button.CANCEL),
[Button.CYCLE_SHINY]: () => this.buttonDirection(Button.CYCLE_SHINY),
[Button.CYCLE_FORM]: () => this.buttonDirection(Button.CYCLE_FORM),
[Button.CYCLE_GENDER]: () => this.buttonDirection(Button.CYCLE_GENDER),
[Button.CYCLE_ABILITY]: () => this.buttonDirection(Button.CYCLE_ABILITY),
[Button.CYCLE_NATURE]: () => this.buttonDirection(Button.CYCLE_NATURE),
[Button.CYCLE_VARIANT]: () => this.buttonDirection(Button.CYCLE_VARIANT),
```
After:
```typescript
[Button.ACTION]: () => this.buttonAb(Button.ACTION),
[Button.CANCEL]: () => this.buttonAb(Button.CANCEL),
[Button.CYCLE_SHINY]: () => this.buttonCycleOption(Button.CYCLE_SHINY),
[Button.CYCLE_FORM]: () => this.buttonCycleOption(Button.CYCLE_FORM),
[Button.CYCLE_GENDER]: () => this.buttonCycleOption(Button.CYCLE_GENDER),
[Button.CYCLE_ABILITY]: () => this.buttonCycleOption(Button.CYCLE_ABILITY),
[Button.CYCLE_NATURE]: () => this.buttonCycleOption(Button.CYCLE_NATURE),
[Button.CYCLE_VARIANT]: () => this.buttonCycleOption(Button.CYCLE_VARIANT),
```

### Screenshots/Videos


Not applicable as the changes do not alter the user interface directly.

## How to test the changes?




To test these changes:
- Run the application and test the affected button actions (`ACTION`, `CANCEL`, `CYCLE_*`) to ensure they function as expected.

## Checklist
